qwandaq/src/main/java/life/genny/qwandaq/utils/FilterUtils.java

```diff
@@ -0,0 +1,412 @@
+
+        if (value.contains(FilterConst.DATETIME)){
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_GREATER_THAN));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_GREATER_THAN_OR_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_LESS_THAN));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_LESS_THAN_OR_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_NOT_EQUAL_TO));
+            return base;
+        } else if (value.contains(FilterConst.SELECT)) {
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_NOT_EQUAL_TO));
+            return base;
+        } else {
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_NOT_EQUAL_TO));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_LIKE));
+            base.add(beUtils.getBaseEntityOrNull(FilterConst.SEL_NOT_LIKE));
+        }
+
+        return base;
+    }
+
+    /**
+     * Return ask with filter select option values
+     *
+     * @param queGrp  Question Group
+     * @param queCode Question code
+     * @param lnkCode Link code
+     * @param lnkVal  Link Value
+     * @return Data message of filter select box
+     */
+    public QDataBaseEntityMessage getSelectBoxValueByCode(String queGrp,String queCode,String lnkCode,String lnkVal) {
+        QDataBaseEntityMessage base = new QDataBaseEntityMessage();
+
+        base.setParentCode(queGrp);
+        base.setLinkCode(FilterConst.LNK_CORE);
+        base.setLinkValue(FilterConst.LNK_ITEMS);
+        base.setQuestionCode(queCode);
+
+        SearchEntity searchBE = new SearchEntity(FilterConst.SBE_DROPDOWN, FilterConst.SBE_DROPDOWN)
+                .add(new Column(PRI_CODE, FilterConst.PRI_CODE_LABEL));
+        searchBE.setRealm(userToken.getProductCode());
+        searchBE.setLinkCode(lnkCode);
+        searchBE.setLinkValue(lnkVal);
+        searchBE.setPageStart(0).setPageSize(1000);
+
+        List<BaseEntity> baseEntities = searchUtils.searchBaseEntitys(searchBE);
+        List<BaseEntity> basesSorted =  baseEntities.stream()
+                .sorted(Comparator.comparing(BaseEntity::getName))
+                .collect(Collectors.toList());
+
+        base.setItems(basesSorted);
+
+        return base;
+    }
+
+    /**
+     * Return ask with bucket filter options
+     * @param sbeCode Search entity code
+     * @param lnkCode Link Code
+     * @param lnkValue Link Value
+     * @return Bucket filter options
+     */
+    public SearchEntity getQuickOptions(String sbeCode,String lnkCode, String lnkValue) {
+        SearchEntity searchBE = new SearchEntity(sbeCode,sbeCode);
+        searchBE.add(new Or(new Filter(FilterConst.PRI_CODE, Operator.LIKE, "CPY_%")
```

Comment:
use `STARTS_WITH` enum and `Prefix.java` constant here instead

Reply:
used Prefix.java